packages/lodestar/src/api/impl/beacon/state/utils.ts

```diff
@@ -2,31 +2,35 @@
 import {GENESIS_SLOT, FAR_FUTURE_EPOCH, CachedBeaconState} from "@chainsafe/lodestar-beacon-state-transition";
 import {phase0} from "@chainsafe/lodestar-beacon-state-transition";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
-import {IForkChoice} from "@chainsafe/lodestar-fork-choice";
-import {Epoch, ValidatorIndex, Gwei, Slot} from "@chainsafe/lodestar-types";
+import {Epoch, ValidatorIndex, Gwei, Slot, Root} from "@chainsafe/lodestar-types";
 import {fromHexString, readOnlyMap} from "@chainsafe/ssz";
 import {IBeaconChain} from "../../../../chain";
-import {StateContextCache} from "../../../../chain/stateCache";
 import {IBeaconDb} from "../../../../db/api";
 import {StateId} from "./interface";
 
+export enum RegenType {
+  CacheOnly,
+  AllowRegen,
+}
+
 export async function resolveStateId(
   chain: IBeaconChain,
   db: IBeaconDb,
-  stateId: StateId
+  stateId: StateId,
+  type: RegenType = RegenType.CacheOnly
```

Comment:
Only the debug state endpoint is able to trigger regen. This is a serious DOS vector so I think it makes that is only allowed for the debug API which is not enabled in normal usage.